app/middleware/service-selection.ts

```diff
@@ -1,7 +1,41 @@
 import { NextFunction, Response } from 'express';
 import { ServiceConfig } from '../models/services/base-service';
 import HarmonyRequest from '../models/harmony-request';
-import { chooseServiceConfig } from '../models/services';
+import { chooseServiceConfig, getServiceConfigs } from '../models/services';
+
+/**
+ * Add collections to service configs
+ * @param req - The client request
+ * @returns An array of service configurations
+ */
+function addCollectionsToServicesByAssociation(req: HarmonyRequest): ServiceConfig<unknown>[] {
+  const configs = getServiceConfigs();
```

Comment:
Would it be useful to push the logic of adding these collections to the configs into getServiceConfigs, e.g. getServiceConfigs(req.collections), where the collections are an optional parameter? 

I'm envisioning some other use cases that might want this same kind of behavior. E.g. if there's ever a web GUI component of Harmony, someone might want to look up which services are available with their collection(s) without actually making a data transformation request. 

What do you think?

Reply:
That's an interesting idea. I can see potential utility, but I also like the idea of keeping it as a simple getter. I'm going to leave it as is for now.